### PR TITLE
Simplify structure

### PR DESCRIPTION
**Pull request by h-mayorquin**
Follow up from https://github.com/catalystneuro/ndx-binned-spikes/pull/16

Uses a single structure instead of two as in the aferomentioned PR.  

A good place to start the review will be to read the documentation:
https://github.com/catalystneuro/ndx-binned-spikes/tree/simplify_structure

### CODE REVIEW

**Comment by bendichter** [comment]
This is looking good. One question I have is: what are the condition_indices indexing? In some cases, maybe 0,1,2,3 makes sense, but I think often it might make more sense to list the conditions. Maybe we could add a `conditions` array that is a list of strings. For example, in the IBL dataset the condition may indicate the different conditions for stimulus constrast. For the DiCarlo datasets, it may indicate the filenames of the unique visual stimuli.

**Comment by h-mayorquin** [comment]
> One question I have is: what are the condition_indices indexing? In some cases, maybe 0,1,2,3 makes sense, but I think often it might make more sense to list the conditions. Maybe we could add a conditions array that is a list of strings. For example, in the IBL dataset the condition may indicate the different conditions for stimulus constrast. For the DiCarlo datasets, it may indicate the filenames of the unique visual stimuli.

When I was re-writing this I was thinking that we could add an extra column for the conditions. I was thinking that we could make that extra column a link to either indexed stimuli or a column of a dynamic table or whatever.

But now, I think what you proposal is simpler and maybe better. Just pass a vector with conditions as string as you mention. Note that with this could also make `condition_indices` an internal implementation detail.

File: README.md [hunk 1]
```diff
@@ -41,12 +41,12 @@ data = np.array(
...
 )
 
-event_timestamps = np.array([0.25, 5.0, 12.25])  # The timestamps to which we align the counts
+timestamps = np.array([0.25, 5.0, 12.25])  # The timestamps to which we align the counts
 milliseconds_from_event_to_first_bin = -50.0  # The first bin is 50 ms before the event
 bin_width_in_milliseconds = 100.0  # Each bin is 100 ms wide
 binned_aligned_spikes = BinnedAlignedSpikes(
     data=data,
-    event_timestamps=event_timestamps,
+    timestamps=timestamps,
```
Comment (bendichter) [inline README.md]: why did you change this? I liked it before. It disambiguates this from the timestamps of the bins

Comment (h-mayorquin) [reply README.md]: Interesting point, I missed that it was useful to disambiguate that way. The reason I changed this was to make it consistent with most pynwb objects that have a timestamp attribute but I can revert.

Comment (bendichter) [reply README.md]: I do think it would make sense to call this variable `timestamps` if the object inherits from `TimeSeries`. I noticed you decided not to do that here. Did you consider inheriting from `TimeSeries`? Why did you make the decision not to. I'm on the fence about it myself so I'm curious what you think.

File: README.md [hunk 2]
```diff
@@ -160,14 +160,14 @@ units_region = DynamicTableRegion(
...
 bin_width_in_milliseconds = 100.0
 name = "BinnedAignedSpikesForMyPurpose"
 description = "Spike counts that is binned and aligned to events."
 binned_aligned_spikes = BinnedAlignedSpikes(
     data=data,
-    event_timestamps=event_timestamps,
+    timestamps=timestamps,
     bin_width_in_milliseconds=bin_width_in_milliseconds,
     milliseconds_from_event_to_first_bin=milliseconds_from_event_to_first_bin,
     description=description,
```
Comment (bendichter) [inline README.md]: this does not have event conditions

File: spec/ndx-binned-spikes.extensions.yaml [hunk 3]
```diff
@@ -37,71 +37,23 @@ groups:
...
     - null
-    doc: The index of the event that each row of the data corresponds to.
-  - name: timestamps
-    dtype: float64
-    dims:
-    - number_of_events
-    shape:
-    - null
-    doc: The timestamps at which the events occurred.
+    doc: The index of the condition that each timestamps corresponds to (e.g. a stimuli
```
Comment (bendichter) [inline spec/ndx-binned-spikes.extensions.yaml]:
```suggestion
    doc: The index of the condition that each timestamps corresponds to (e.g. a stimulus
```

File: README.md (hunk 1 above)
Comment (h-mayorquin) [reply README.md]: I thought about it but it has some attributes that I don't think make a lot of sense here. continuity, conversion, offset, etc. I think in some cases it might make sense to have a rate instead of timestamps but that also seemed like a complication.

File: README.md (hunk 2 above)
Comment (h-mayorquin) [reply README.md]: Yeah, I introduce the possibility of storing multiple conditions in the next section. I think it makes presentation easier if less concepts are introduced at the same time.

File: spec/ndx-binned-spikes.extensions.yaml (hunk 3 above)
Comment (h-mayorquin) [reply spec/ndx-binned-spikes.extensions.yaml]: Thanks.

File: README.md (hunk 1 above)
Comment (bendichter) [reply README.md]: yeah I would agree, let's not use `TimeSeries`. I think that is a good reason to use `event_timestamps`, so this isn't mistaken for a `TimeSeries`

**Comment by bendichter** [comment]
I really think it could use an optional argument `condition_labels`, which are string labels for each of the `condition_indices`. I think this is particularly true if you are going to use the term "indices". If you use that term, then those integers must be indexing something. If you add a `condition_labels`, then this is the dataset that is being indexed.

File: README.md (hunk 1 above)
Comment (h-mayorquin) [reply README.md]: OK, let me go back to this.

**Comment by h-mayorquin** [comment]
> I really think it could use an optional argument `condition_labels`, which are string labels for each of the `condition_indices`. I think this is particularly true if you are going to use the term "indices". If you use that term, then those integers must be indexing something. If you add a `condition_labels`, then this is the dataset that is being indexed.

Let's do this!

File: README.md (hunk 1 above)
Comment (h-mayorquin) [reply README.md]: Changed.

**Comment by h-mayorquin** [comment]
@bendichter this should be ready. I separated the condition_labels request in #18 to simplify the discussion. Let me know what you think.

**Review by bendichter — APPROVED** [review]
